client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -568,9 +576,12 @@ private static HttpRequestBase createHttpRequest(String method, URI uri, HttpEnt
         }
     }
 
-    private static HttpRequestBase addRequestBody(HttpRequestBase httpRequest, HttpEntity entity) {
+    private HttpRequestBase addRequestBody(HttpRequestBase httpRequest, HttpEntity entity) {
```

Comment:
This can remain static and receive the `comporessionEnabled` flag as parameter

Reply:
That's what I initially did, but it required to change all call sites to add `compressionEnabled` as a paramter  and this felt wrong since it's a class field. Since both `createHttpRequest` and `addRequestBody` are private, making them instance methods should be ok as they now need to access instance data.

Reply:
It's a minor thing (especially as you mention, they are private methods), but I think the methods (should you choose to keep them static functions) will be more expressive in terms of what they're meant to do and what influences their behaviour.

Reply:
I see your point even if passing a class field through 2 methods calls to keep them static feels a bit verbose. Fixed in fe4b90f